azurerm/resource_arm_private_dns_cname_record.go

```diff
@@ -0,0 +1,167 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPrivateDnsCNameRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPrivateDnsCNameRecordCreateUpdate,
+		Read:   resourceArmPrivateDnsCNameRecordRead,
+		Update: resourceArmPrivateDnsCNameRecordCreateUpdate,
+		Delete: resourceArmPrivateDnsCNameRecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
could we add some validation to this? e.g.

```suggestion
				ForceNew: true,
				ValidateFunc: validate.NoEmptyStrings,
```